pkg/querier/stats/query_metrics.go

```diff
@@ -52,6 +55,11 @@ func NewQueryMetrics(reg prometheus.Registerer) *QueryMetrics {
 			Name:      "querier_queries_rejected_total",
 			Help:      "Number of queries that were rejected, for example because they exceeded a limit.",
 		}, []string{"reason"}),
+		QueriesExecutedTotal: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Namespace: "cortex",
+			Name:      "querier_queries_executed_total",
+			Help:      "Number of queries that were executed against a particular source.",
```

Comment:
should we clarify that this refers to promQL queries (as opposed to gRPC queries to an ingester or a store-gateway)? Also the sum of cortex_querier_queries_executed_total will be larger than the actual promQL queries executed because one query can request from both the store-gateway and the ingester.

How about naming the metric and the field something along the lines of  `cortex_querier_queries_storage_type_total{storage="ingester|store-gateway"}`?

Reply:
Good suggestions.